src/app/ReadClient.h

```diff
@@ -165,6 +165,25 @@ class ReadClient : public Messaging::ExchangeDelegate
          * SendAutoResubscribeRequest is not called, this function will not be called.
          */
         virtual void OnDeallocatePaths(ReadPrepareParams && aReadPrepareParams) {}
+
+        /**
+         * This function is invoked when constructing a read/subscribeRequest that does not have data
+         * version filters specified, to give the callback a chance to provide some.
+         *
+         * This function is expected to encode as many complete data version filters as will fit into
+         * the buffer, rolling back any partially-encoded filters if it runs out of space, and return
+         * the boolean to show if it has successfully encode at least one data version filter.
+         */
+        virtual CHIP_ERROR OnUpdateDataVersionFilterList(DataVersionFilterIBs::Builder & aDataVersionFilterIBsBuilder,
+                                                         const Span<AttributePathParams> & aAttributePaths,
+                                                         bool & aHasEncodeDataVersionList)
+        {
+            return CHIP_NO_ERROR;
```

Comment:
Should set `aEncodedDataVersionList` to false here, and in other implementations as needed.